src/Vault.sol

```diff
@@ -144,44 +145,56 @@ contract Vault is DefaultAccessControl, IERC721Receiver {
     uint256 public globalStabilisationFeePerUSDSnapshotD = 0;
 
     /// @notice Creates a new contract
-    /// @param admin Protocol admin
     /// @param positionManager_ UniswapV3 position manager
     /// @param factory_ UniswapV3 factory
     /// @param protocolGovernance_ UniswapV3 protocol governance
-    /// @param oracle_ Oracle
     /// @param treasury_ Vault fees treasury
-    /// @param stabilisationFee_ MUSD initial stabilisation fee
     constructor(
-        address admin,
         INonfungiblePositionManager positionManager_,
         IUniswapV3Factory factory_,
         IProtocolGovernance protocolGovernance_,
-        IOracle oracle_,
         address treasury_,
-        uint256 stabilisationFee_
-    ) DefaultAccessControl(admin) {
+        address token_
+    ) {
         if (
             address(positionManager_) == address(0) ||
             address(factory_) == address(0) ||
             address(protocolGovernance_) == address(0) ||
-            address(oracle_) == address(0) ||
-            address(treasury_) == address(0)
+            address(treasury_) == address(0) ||
+            address(token_) == address(0)
         ) {
             revert AddressZero();
         }
 
+        positionManager = positionManager_;
+        factory = factory_;
+        protocolGovernance = protocolGovernance_;
+        treasury = treasury_;
+        token = IMUSD(token_);
```

Comment:
Let's also set `initialized` to `true` here. This will ensure implementation contract can only be used through the proxy contract.